d2layouts/d2sequence/sequence_diagram.go

```diff
@@ -402,19 +402,18 @@ func (sd *sequenceDiagram) placeSpans() {
 		// finds the position if there are messages to this span
 		minMessageY := math.Inf(1)
 		if firstMessage, exists := sd.firstMessage[span]; exists {
-			// needs to check Src/Dst because of self-edges or edges to/from descendants
-			if span == firstMessage.Src {
+			if firstMessage.Src == firstMessage.Dst || span == firstMessage.Src {
```

Comment:
why don't u just do like get min and max of all points instead of assuming src/dst? seems less robust

Reply:
Edges to/from descendants need this, as well self edges.
In these cases, taking the min/max of the first/last points will make spans bigger than they they should be

This is what happens to `sequence_diagram_self_edges` if
```
minMessageY := math.Inf(1)
		if firstMessage, exists := sd.firstMessage[span]; exists {
			minMessageY = math.Min(firstMessage.Route[0].Y, firstMessage.Route[len(firstMessage.Route)-1].Y)
		}
		maxMessageY := math.Inf(-1)
		if lastMessage, exists := sd.lastMessage[span]; exists {
			maxMessageY = math.Max(lastMessage.Route[len(lastMessage.Route)-1].Y, lastMessage.Route[0].Y)
		}
```

Result
![image](https://user-images.githubusercontent.com/102628724/206252879-3a48ff93-1276-43e3-bb77-89465edc8115.png)

Expected
![image](https://user-images.githubusercontent.com/102628724/206253133-02871e26-b421-4994-8630-b0a4326aff68.png)